org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -1678,7 +1678,7 @@ class Diagnostic {
 	/**
 	 * The diagnostic's code. Can be omitted.
 	 */
-	String code
+	Either<String, Number> code
```

Comment:
@cdietrich said:

> do we have any general criteria when we turn number into Integer?

Reply:
also: should we offer a "comfort-constructor" with Integer or Long or Number?

Reply:
Yes, we should - as adding that does not break API it can be added more easily and as I am not sure what is best needed here I will leave the general case for now.